src/lib/validator.ts

```diff
@@ -0,0 +1,44 @@
+import Ajv from 'ajv'
+
+
+function validate(schema: any, data = {}) {
+    const ajv = new Ajv({})
+    const valid = ajv.compile(schema);
+    if (!valid(data)) {
+        return valid.errors![0];
+    } 
+}
+
+const userSchema = {
+  type: 'object',
+  properties: {

```

Comment:
I found there are validation rules on the these fields, can I find the rules in the API document, API response (when validation fails), or test cases?

Reply:
my api docs might be way too unprofessional. I think there is a validation case in test 'naively' demonstrates such check, in a rather simpler way, the request will end up with a predefined errCode. 

test('login json validates', async () => {
    const res = await server
        .post('/user')
        .send({
            userName: '123', // username contains unexpected combinations
            password: 'a', // min length violation
        })
    expect(res.body.errCode).not.toBe(-1)
})